templates/search/search_clients_daily_v8/query.sql

```diff
@@ -29,6 +29,19 @@ CREATE TEMP FUNCTION get_search_addon_version(active_addons ANY type) AS (
   )
 );
 
+-- Take array of experiment structs and return union of experiments.key_value arrays
+-- This is a workaround for the inability to use ARRAY_CONCAT_AGG as an analytical function
+CREATE TEMP FUNCTION union_experiments(list ANY TYPE) AS (
```

Comment:
This is a kinda ugly function to get around not being able to use `ARRAY_CONCAT_AGG` over a window.  It does a union of the experiments by concatenating all the array and then deduping it.  Is there an easier way to do this that I'm missing?

Reply:
I would recommend this:
```sql
CREATE TEMP FUNCTION udf_map_mode_last(list_of_maps ANY TYPE) AS (
  ARRAY(
    SELECT AS STRUCT
      key,
      udf_mode_last(ARRAY_AGG(value)) AS value
    FROM
      UNNEST(list_of_maps),
      UNNEST(key_value)
    GROUP BY
      key
  )
);
...
    udf_map_mode_last(ARRAY_AGG(experiments) OVER w1) AS experiments,
```

Reply:
That makes more sense